lxc/cookiejar/jar_test.go

```diff
@@ -130,7 +133,7 @@ func (s *cookieJarSuite) TestJar() {
 	require.NoError(err)
 
 	// Should open without error.
-	j2, err := Open(s.j.filepath)
+	j2, err := Open(s.j.filepath, s.host)
```

Comment:
What we're aiming for in this work is to prevent the cookie jar from saving cookies that are not from the specified remote. So for this test, the cookiejar should only be saving cookies from `fakehost` - as the test currently stands we want it to fail.

Reply:
I moved the domain check from jar creation to when the cookies are set, which allows for the expected test behavior. Now, the cookie jar is created with the configured remote address, and each SetCookie checks the incoming domain with the domain of the remote, and only sets the cookie when they match. This is in addition to the PublicSuffixList check, which (contrary to my initial understanding) functions as a deny list for any cookies that have a domain maliciously set to ".com", ".org", etc.

I've added a test for the SetCookie logic, in addition to modifying the cookies in the TestJar test, to include two cookies coming from wrong domains that should be ignored. PTAL at that new test setup.